src/main/java/hudson/plugins/git/extensions/impl/CloneOption.java

```diff
@@ -157,13 +161,75 @@ public void decorateFetchCommand(GitSCM scm, GitClient git, TaskListener listene
         cmd.timeout(timeout);
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public GitClientType getRequiredClient() {
         return GitClientType.GITCLI;
     }
 
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        CloneOption that = (CloneOption) o;
+
+        if (shallow != that.shallow) {
+            return false;
+        }
+        if (noTags != that.noTags) {
+            return false;
+        }
+        if (depth != that.depth) {
+            return false;
+        }
+        if (honorRefspec != that.honorRefspec) {
+            return false;
+        }
+        if (reference != null ? !reference.equals(that.reference) : that.reference != null) {
+            return false;
+        }
+        return timeout != null ? timeout.equals(that.timeout) : that.timeout == null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        return CloneOption.class.hashCode();
```

Comment:
The hashcode must be consistent or the object cannot be found again. Thus you must exclude mutable fields from the hashCode calculation.

If all fields are mutable then no fields may be used to calculate the hashCode.

Since we expect these to end up in collections with their sibling class implementations and we only ever have at most one impl of a given type, we just give each impl a different constant hashCode and all is right with the world